### PR TITLE
Add toybox and products resources

### PR DESCRIPTION
**Pull request by yasgreen93**
# Description

- Adding resource for toybox
- Adding resource for products.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have made my code self-documenting where possible, with the use of clear variable and function names
- [ ] I have commented the reasons for any complex or unusual code, so others can understand the context
- [x] I have made corresponding changes to the documentation
- [ ] I have added relevant tests for any new functionality I have added
- [x] I have updated the CHANGELOG
- [x] My changes do not generate any new warnings, either in tooling or in the browser console

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,11 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+## [0.0.8] - 2020-04-28
+### 🚀 Features
+- Adding resource for toybox to designate and undesignate toy for return.
+- Adding resource for products get products with query params.
+
```
Comment (Jamiewarb) [inline CHANGELOG.md:13]: Please add this under 'unreleased' as it's not released yet. When merging from develop to master, we add a release tag and date

```suggestion
## [Unreleased]
### 🚀 Features
- Adding resource for toybox to designate and undesignate toy for return.
- Adding resource for products get products with query params.

```

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "whirli-client",
-  "version": "0.0.7",
+  "version": "0.0.8",
```
Comment (Jamiewarb) [inline package.json:3]: Please only change this version on the PR from develop to master, when the code gets released

File: src/Client.ts [hunk 3]
```diff
@@ -8,7 +8,9 @@ import BasketLine from './Resources/Basket/BasketLine';
 import Subscriptions from './Resources/Subscriptions/Subscriptions';
 import SubscriptionAddons from './Resources/SubscriptionAddons/SubscriptionAddons';
 import Codes from './Resources/Codes/Codes';
+import Toybox from './Resources/Toybox/Toybox';
 import { loadWaccResources, WaccResources } from './Resources/Wacc';
+import Products from './Resources/Products/Products';
```
Comment (Jamiewarb) [inline src/Client.ts:13]: Please add these resources to the Client test file as well

File: src/Resources/Products/Products.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
+import AbstractResource from '../AbstractResource';
+
+export default class Products extends AbstractResource {
+    initialise(): void {
+        this.resourcePath = '/products';
+        this.defaultAccess = 'guest';
+    }
+
+    /**
+     * **Main endpoint for browse toys.**
```
Comment (Jamiewarb) [inline src/Resources/Products/Products.ts:10]: Comment is malformed

Comment (yasgreen93) [reply src/Resources/Products/Products.ts:10]: It's meant to be like that, markdown to make it bold.
Screenshot 2020-04-28 at 16 04 09

Comment (Jamiewarb) [reply src/Resources/Products/Products.ts:10]: Very interesting, thanks for showing me that

File: src/Resources/Products/Products.ts [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+     * @param {string} page - pagination.
+     * @param {string} search - Search term.
+     * @param {string} match - Returns all records that match the query, for example `productCategories.name:board%20games`.
+     * @param {string} orderBy - Field you want to order by, for example `created_at`.
+     * @param {string} sortedBy - Ascending or descending, needs `orderBy` field - `asc` or `desc`.
+     * @param {string} to - Anything equal or smaller than column:number, for example `tokens:30`.
+     * @param {string} from - Anything equal or large than column:number, for example `tokens:1`.
+     */
+    public get: Function = this.createMethodFromPartialSpec({
+        method: 'POST',
```
Comment (Jamiewarb) [inline src/Resources/Products/Products.ts:21]: This this may need to be type `GET`

**Pull request merged**